### PR TITLE
Search, Join tournament beginnings

### PR DESCRIPTION
**Pull request by williambertrand**
### Description
In scope of this PR is just searching for a tournament, still not exactly sure how we want to handle actually joining a tournament. Do we show the Tournament details page with all the brackets? Do we just show the brackets? 

| Create a tournament | Searching |
| --- | --- |
|  |  |

### CODE REVIEW

File: components/Tournaments/TournamentList.tsx [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
-const TournamentList = ({ tournaments, navigation }: TournamentListProps) => {
+const TournamentList = ({ tournaments, navigation, emptyStateText, onPressTournament }: TournamentListProps) => {
 
     const renderItem = ({item}: {item: Tournament}) => {
         return (
-            <View style={styles.row}>
-                <View style={styles.left}>
-                    <Text style={styles.nameText}>{item.name}</Text>
+            <TouchableHighlight style={styles.row} activeOpacity={0.4} underlayColor={'#fff'} onPress={() => onPressTournament(item)}>
+                <View>
```
Comment (williambertrand) [inline components/Tournaments/TournamentList.tsx:19]: I always forget, `TouchableHighlight` can only have 1 child

Also, styling and onPress styling needs some work, but we can deal with that later

File: screens/TournamentDiscovery/JoinTournamentScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      setResults(results);
+      setLoading(false);
+    } catch (e) {
+      // TODO: Display error toast, but for now, just log
+      console.log(`Error searching for tournament: ${e}`);
+    }
+  };
+
+  const joinTournament = (tournament: Tournament) => {
+    // TODO: Navigate to tournament details page for joining?
```
Comment (probicheaux) [inline screens/TournamentDiscovery/JoinTournamentScreen.tsx:45]: yeah, that makes the most sense to me

File: screens/TournamentDiscovery/JoinTournamentScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      <Text style={styles.title}>Join Tournament</Text>
+
+      <Text style={styles.label}>Search by name</Text>
+      <TextInput
+        value={data.name}
+        style={styles.input}
+        onChangeText={(n) => setData({ ...data, name: n })}
+        returnKeyType="search"
+        placeholder="Tournament Name"
+        onSubmitEditing={submit}
```
Comment (probicheaux) [inline screens/TournamentDiscovery/JoinTournamentScreen.tsx:85]: nice, this is coool

**Review by probicheaux — APPROVED** [review]
dope, lgtm

**Pull request merged**